src/main/java/me/youhavetrouble/purpurextras/listeners/FarmingListener.java

```diff
@@ -0,0 +1,54 @@
+package me.youhavetrouble.purpurextras.listeners;
+
+import me.youhavetrouble.purpurextras.PurpurExtras;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.block.data.type.Cocoa;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class FarmingListener implements Listener {
+
+    List<Material> farmables = Arrays.asList(Material.CARROTS, Material.COCOA, Material.NETHER_WART, Material.POTATOES, Material.WHEAT, Material.BEETROOTS);
```

Comment:
this can be configurable, get a string list from config, toUppercase the entries, match with material enum, then check if it matches with this list